tests/test_decap.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+from netaddr import *
+import time
+import logging
+import requests
+from ptf_runner import ptf_runner
+from ansible_host import AnsibleHost
+from datetime import datetime
+
+
+
+
+def decap_entry_cfg_swss(duthost, decap_setup, outer_ipv4, outer_ipv6, op):
+    testbed_type, lo_ipv6, lo_ipv4,dscp_mode, ecn_mode,ttl_mode = decap_setup
+    dst_ip = lo_ipv4 if outer_ipv4 is True else lo_ipv6
+    extra_vars = {
+        'outer_ipv4': outer_ipv4,
+        'outer_ipv6': outer_ipv6,
+        'lo_ip': lo_ipv4,
+        'lo_ipv6': lo_ipv6,
+        "dscp_mode": dscp_mode,
+        "ecn_mode": ecn_mode,
+        "ttl_mode": ttl_mode,
+        "op": op
+    }
+
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+    duthost.template(src="decap/decap_conf.j2", dest="/tmp/decap_conf.json")
+    duthost.shell('docker cp /tmp/decap_conf.json swss:/decap_conf.json')
+    duthost.shell('docker exec swss sh -c "swssconfig /decap_conf.json"')
+
+def decap_config_ptfhost(ptf_vars,ptfhost):
+    ptfhost.host.options['variable_manager'].extra_vars.update(ptf_vars)
+    logging.info("extra_vars: %s" % str(ptfhost.host.options['variable_manager'].extra_vars))
+    ptfhost.template(src="fib/fib.j2", dest="/root/fib_info.txt")
+    ptfhost.copy(src="ptftests", dest="/root")
+
+
+@pytest.fixture(scope="module")
+def decap_setup(ansible_adhoc, testbed,duthost,ptfhost):
+    hostname = testbed['dut']
+    testbed_type = testbed['topo']['name']
+
+    props= testbed['topo']['properties']['configuration_properties']['spine']
+    props_tor = testbed['topo']['properties']['configuration_properties']['tor']
+
+    ans_host = AnsibleHost(ansible_adhoc, hostname)
```

Comment:
Suggest to remove:
* Fixture `ansible_adhoc`. The `duthost` and `ptfhost` fixtures are enough.
* Variables: hostname, props, pros_tor, ans_host